### PR TITLE
seo(website): build-time tech-name autolinking via remark plugin

### PR DESCRIPTION
**Pull request by rawkode**
## Why

Articles mention technologies in prose constantly but only link to \`/technology/{id}\` when the author remembers to write the link. Result: most technology profile pages are link-orphans (Tier-2 SEO audit flagged inbound counts under 3 on the majority of profiles), and articles miss internal-linking signal that compounds across the site.

## What

A remark plugin that runs at build time over every MDX article. On the FIRST plain-text mention of any known technology name, the text is rewritten into a link to \`/technology/{id}\`. Subsequent mentions stay as text — one anchor per article per tech is the standard SEO heuristic.

### Files

- \`src/lib/load-tech-lookup.ts\` — sync helper that walks \`content/technologies/\`, reads frontmatter \`name\`, and builds a \`Map<lowercased-name, technology-id>\`. Handles both \`{id}.mdx\` and \`{id}/index.mdx\` layouts.
- \`src/lib/remark-tech-autolink.ts\` — the plugin. Custom alnum-and-hyphen identifier boundaries so \`kube-vip\` and \`cert-manager\` match correctly, while \`apko-style\` or \`x-apko-y\` do not. Skips code blocks, inline code, headings, existing links, raw HTML, and MDX/JSX components.
- \`src/tests/remark-tech-autolink.test.ts\` — 15 unit tests covering: first-mention linking, subsequent-mention skip, case preservation, hyphenated names, code-block skip, heading skip, link skip, default skip list (Go), no-autolink HTML-comment opt-out, multi-tech in one paragraph, substring rejection (alnum-bounded), hyphen-bounded rejection, custom \`skipNames\` override, JSX/MDX skip.
- \`astro.config.mts\` — wired into \`markdown.remarkPlugins\` with the lookup loaded once at config init.

### Curated skip list

False-positive risk words that exist as technology IDs but are too ambiguous in prose: \`go\`, \`ko\`, \`d\`, \`k\`, \`bun\`, \`act\`, \`lit\`, \`next\`. These are NEVER auto-linked; existing manual links on them are unaffected. Authors can extend the skip list per-call if needed.

### Per-file opt-out

Add the literal HTML comment \`<!-- no-autolink -->\` anywhere in the article body to disable the plugin for that file.

### Cost

Build-time only. One \`readdir\` of the technologies folder at config load, then one regex pass per text node. With ~345 tech profiles and ~14 articles, this is a few hundred milliseconds added to the build. Negligible.

## Tested

- 15 unit tests with vitest (workspace install is broken locally; CI will validate).
- Adversarial review identified the hyphen-boundary edge case (\`apko-style\` was matching pre-fix). Fixed by extending the boundary character class to include \`-\`. Added two regression tests.

## Future work (separate PRs)

- An equivalent rehype pass for raw-HTML content (currently skipped because that content is already author-controlled).
- A link-graph audit script per Tier-2 plan to identify the next set of orphan pages.

This PR was created with the assistance of a LLM.

### CODE REVIEW

File: projects/rawkode.academy/website/src/lib/remark-tech-autolink.ts [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+		// Re-create the global regex per document so the per-text `lastIndex`
+		// state doesn't leak across documents.
+		const globalPattern = new RegExp(pattern.source, "gi");
+		const linked = new Set<string>();
+
+		visit(tree, "text", (node: Text, index, parent) => {
+			if (index === undefined) return;
+			if (shouldSkipParent(parent as LinkableTextParent | undefined)) {
+				return SKIP;
+			}
```
[the remark this answers is not in the document]
Comment (rawkode) [reply projects/rawkode.academy/website/src/lib/remark-tech-autolink.ts]: Resolved by the rewrite to plain recursion (commit e86dd59a). `walk()` checks IGNORE_TYPES before descending, so any subtree rooted at a heading/link/code/MDX-JSX node is never visited. Text inside `[**Kubernetes**](...)` is unreachable by construction, not by per-text-node ancestor checking. The test 'does not link inside an existing link' covers this.

File: projects/rawkode.academy/website/src/lib/remark-tech-autolink.ts [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+			if (!didReplace) return;
+			if (cursor < original.length) {
+				segments.push({ type: "text", value: original.slice(cursor) });
+			}
+
+			const replacement = segments as Text[];
+			(parent as Parent).children.splice(index, 1, ...replacement);
+			return [SKIP, index + replacement.length];
```
[the remark this answers is not in the document]
Comment (rawkode) [reply projects/rawkode.academy/website/src/lib/remark-tech-autolink.ts]: Resolved by the rewrite. The current implementation uses a plain `Node[]` array (no `Text[]` cast) and the splice target is typed `Node[]` too, so no unsafe cast is introduced at the splice site.

File: projects/rawkode.academy/website/astro.config.mts [hunk 3]
```diff
@@ -228,6 +230,20 @@ export default defineConfig({
 		checkOrigin: true,
 	},
 	markdown: {
+		// `CONTENT_TECH_DIR` resolves to the `technologies/` subdirectory when
+		// it exists; otherwise it falls back to the content package root.
+		// Either way, the loader needs the technologies folder itself.
+		remarkPlugins: CONTENT_TECH_DIR
```
[the remark this answers is not in the document]
Comment (rawkode) [reply projects/rawkode.academy/website/astro.config.mts]: Addressed in commit a8595905. Added a configurable `skipPathSubstrings` option to the plugin, defaulting to `['/content/technologies/']`. The transform now reads `file.path` (or `file.history[0]`) and bails before walking when any deny substring matches. Two new tests cover the scoping (tech profile path is skipped; article path still links).
